drivers/vmwarevsphere/vm.go

```diff
@@ -293,3 +293,56 @@ func (d *Driver) addCustomAttributes(vm *object.VirtualMachine) error {
 
 	return nil
 }
+
+func (d *Driver) resizeDisk(vm *object.VirtualMachine) error {
+	devices, err := vm.Device(d.getCtx())
+	if err != nil {
+		return err
+	}
+
+	var disks []*types.VirtualDisk
+	for _, device := range devices {
+		switch md := device.(type) {
+		case *types.VirtualDisk:
+			disks = append(disks, md)
+		default:
+			continue
+		}
+	}
+
+	if len(disks) < 1 {
+		return fmt.Errorf("No disks found for vm: %s", vm.InventoryPath)
+	}
+
+	// only allow an edit the first disk in the order
+	editdisk := disks[0]
+	newSize := int64(d.DiskSize) * 1024
+	if newSize <= editdisk.CapacityInKB {
+		log.Infof("Can only resize up, passed size is less than or equal to the cloned disk size: %dKb <= %dKb",
+			newSize, editdisk.CapacityInKB)
+		return nil
+	}
+
+	log.Infof("Resizing disk up from %dKb to %dKb",
```

Comment:
Should we log identifying information here to enable "something went wrong I'm looking through the logs" sanity checks.